From d07087108602502505d37624c31f1553bea94ee6 Mon Sep 17 00:00:00 2001
From: David Kershner <dkershner@habana.ai>
Date: Thu, 12 Jun 2025 23:47:40 +0300
Subject: [PATCH] [SW-226950] oobi: oobi_sensor_op_state enum

SetNumericSensorEnable persists just the DISABLED and UNAVAILABLE
states. Use a local enum to convert from pldm_sensor_op_state_t to a
single bit.

Change-Id: If0e8436b48b42dfb16680a210e897092747810d1
Signed-off-by: David Kershner <dkershner@habana.ai>
---

diff --git a/oobi.h b/oobi.h
index b07ebfe..93f0726 100644
--- a/oobi.h
+++ b/oobi.h
@@ -38,6 +38,11 @@
 #define OOBI_PLDM_MAX_NUM_NUMERIC_SENSORS	300
 #define OOBI_PLDM_MAX_NUM_STATE_SENSORS		64
 
+typedef enum {
+	OOBI_SENSOR_OP_STATE_DISABLED = 0,
+	OOBI_SENSOR_OP_STATE_UNAVAILABLE = 1,
+} oobi_sensor_op_state_t;
+
 struct hl_oobi_pldm_fu_component_state {
 	uint8_t state;
 	uint8_t pass_component_response_code;
@@ -102,7 +107,7 @@
 } __packed;
 
 struct hl_oobi_pldm_numeric_sensor_op_state {
-	uint16_t sensor_op_state:1;
+	uint16_t sensor_op_state:1; /* oobi_sensor_op_state_t */
 	uint16_t sensor_id:15;
 } __packed;
 
